app/views/requests_interests.py

```diff
@@ -0,0 +1,35 @@
+from django.http import Http404
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from ..models import RequestInterest
+from ..serializers import RequestInterestSerializer
+
+
+class RequestInterestList(generics.ListCreateAPIView):
+    permission_classes = (IsAuthenticated,)
+    queryset = RequestInterest.objects.all()
+    serializer_class = RequestInterestSerializer
+
+
+class MentorRequestInterest(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get_object(self, pk):
+        try:
+            return RequestInterest.objects.filter(request_id__exact=pk)
+        except RequestInterest.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk):
+        interest = self.get_object(pk)
+        serializer = RequestInterestSerializer(interest, many=True)
+        return Response(serializer.data)
+
+
+class RequestInterestDetail(generics.RetrieveUpdateDestroyAPIView):
```

Comment:
Shouldn't this get a single `RequestInterest`? Why aren't we passing in a primary key and going a get instead of a get all

Reply:
the generics class-based view is a shorter version of APIView, does the exact same thing as well. Also, with the APIView, I was unable to run the put or patch operation, so i switched to this. It gets a single requestinterest by the way, the retrieve part handles that.